### PR TITLE
Add placeholders directly in the MainTabBarActivity

### LINKED ISSUE #4423: Android Bottom Bar: Create placeholders for pages
The bottom bar is cool, but switching pages can be a bit sluggish, specially on the Reports view and most likely the calendar page when it's ready.

We can add a fade transition to reduce a bit of the problem, but ideally, we would need placeholders to show while the data is loading.

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This adds placeholders directly in the `MainTabBarActivity`.

### Relationships
Closes #4423 

## Why do we want this?
Because our load times aren't fast enough to not require any placeholders.

Why directly on the `MainTabBarActivity`? 
- It takes some time to load even the first fragments.

Why the other fragments don't have placeholders yet?
- They can be done in other PRs or not done at all. The Reports for example doesn't look like it needs one.
- ~Still validating the idea~ As shown internally, the calendar also doesn't seem to need the placeholder, it appears and disappears too fast for it to look good.

## How is it done?
By adding the placeholders directly in the main tab bar activity and handling their visibility based on the "readiness" of each fragment.

### Why not another way?
Feel free to suggest something different or completely refuse this idea (don't do it if you can't suggest something better though).

### Side effects
None.

## Review hints
- Review by commit.
- Do test functionally. I did test, but I'm asking for more tests.

## :squid: Permissions
🦑 it into develop ~after #7021 is merged or directly into #7021~
~This PR is currently targeting #7021 to make the reviews easier.~

### CODE REVIEW

**Comment by daividssilverio** [comment]
Also, we might or might not need to do #7053 depending on the results of this PR.

**Comment by daividssilverio** [comment]
The last two commits also fix a bug in develop that makes android crash when changing configurations (enabling/disabling dark mode).

File: Toggl.Droid/Activities/MainTabBarActivity.ViewBinds.cs [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 using Toggl.Droid.Extensions;
 using Google.Android.Material.BottomNavigation;
+using Toggl.Droid.Fragments;
 
 namespace Toggl.Droid.Activities
 {
     public sealed partial class MainTabBarActivity
     {
         private BottomNavigationView navigationView;
+        private Dictionary<Type, bool> readyLayouts = new Dictionary<Type, bool>();
```
Comment (pdrj) [inline Toggl.Droid/Activities/MainTabBarActivity.ViewBinds.cs]: I don't know if that is a c# thing, but I'd normally use a set if the values are not used.

File: Toggl.Droid/Activities/MainTabBarActivity.cs [hunk 2]
```diff
@@ -238,16 +235,16 @@ private async Task showFragment(int fragmentId)
...
             SupportFragmentManager.ExecutePendingTransactions();
 
             var initialFragment = await getCachedFragment(initialTabItemId);
+            setPlaceholderVisibility(typeof(MainFragment),  initialTabItemId == Resource.Id.MainTabTimerItem && !initialFragment.IsAdded);
             if (!initialFragment.IsAdded)
             {
                 SupportFragmentManager
                     .BeginTransaction()
-                    .Add(Resource.Id.CurrentTabFragmmentContainer, initialFragment, initialTabItemId.ToString())
+                    .Replace(Resource.Id.CurrentTabFragmmentContainer, initialFragment, initialTabItemId.ToString())
```
Comment (pdrj) [inline Toggl.Droid/Activities/MainTabBarActivity.cs]: Could you explain a bit why is there a replace now?

File: Toggl.Droid/Activities/MainTabBarActivity.ViewBinds.cs (hunk 1 above)
Comment (heytherewill) [reply Toggl.Droid/Activities/MainTabBarActivity.ViewBinds.cs]: Agreed

File: Toggl.Droid/Activities/MainTabBarActivity.cs (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Droid/Activities/MainTabBarActivity.cs]: Basically, a crash was going on when the view was being restored after being killed by the system for some reason.
After some trials, that fixed it. This makes sure if any existing fragments with same Id get replaced with the new fragment.

File: Toggl.Droid/Activities/MainTabBarActivity.ViewBinds.cs (hunk 1 above)
Comment (daividssilverio) [reply Toggl.Droid/Activities/MainTabBarActivity.ViewBinds.cs]: Sure. Done.

**Pull request merged**